File: B_Following_the_String.py
t = int(input())  # number of test cases

for _ in range(t):
    n = int(input())  # length of the lost string
    trace = [int(a) for a in input().split()]

    dic = {0 : ['a']}
    char = 'a'
    ans = 'a'
    for num in trace[1:]:
        if num == 0:
            char = chr(ord(char)+1)
            ans += char
            dic[0].append(char)
        else:
            ans += dic[num-1][0]
            try:
                dic[num] += [dic[num-1][0]]
                dic[num-1] = dic[num-1][1:]
            except:
                dic[num] = [dic[num-1][0]]
                dic[num-1] = dic[num-1][1:]
    print(ans)

## copied from codeforces NICE NICE NICE
# t = int(input())
# for _ in range(t):
# 	n = int(input())
# 	l = list(map(int,input().split()))
# 	arr = [-1]*26
# 	s = ""
# 	for i in range(n):
# 		for j in range(26):
# 			if arr[j] == l[i]-1:
# 				arr[j]+=1
# 				s += chr(97+j)
# 				break
# 	print(s)

File: B_Following_the_String.cpp
#include <iostream>
#include <vector>
#include <deque>

using namespace std;

int main() {
    int t;
    cin >> t; // number of test cases

    for (int i = 0; i < t; ++i) {
        int n;
        cin >> n; // length of the lost string

        vector<int> trace(n);
        for (int j = 0; j < n; ++j) {
            cin >> trace[j]; // trace of the string
        }

        vector<deque<char> > dic(1, deque<char>(1, 'a'));
        char ch = 'a';
        string ans = "a";
        for (int j = 1; j < n; ++j) {
            if (trace[j] == 0) {
                ch = ch + 1;
                ans += ch;
                dic[0].push_back(ch);
            } else {
                if (!dic[trace[j] - 1].empty()) {
                    ans += dic[trace[j] - 1][0];
                    dic[trace[j]].push_back(dic[trace[j] - 1][0]);
                    dic[trace[j] - 1].pop_front();
                }
            }
        }
        cout << ans << endl;
    }

    return 0;
}
